### ISSUE TITLE
radial velocity changes with normal vector

### ISSUE THREAD

**Issue by brittonsmith** [yt-project/yt#2012]
### Bug report

Velocity magnitudes in spherical and cylindrical coordinates seem to be dependent on the normal vector, at least if the normal vector is not fully aligned with a cartesian axis. In addition, the spherical radial velocity itself seems dependent on direction of the normal vector under the same conditions.

This can be reproduced by adding the following line to `yt/fields/tests/test_vector_fields.py`:
```
ad.set_field_parameter('normal', np.ones(3) / np.sqrt(3))
```

This will trigger a failure in the test comparing velocity magnitude in all three coordinate systems. As I mentioned above, setting the normal to be entirely in the x, y, or z axis does not result in a test failure. To be clear, this is not roundoff error; the answers are not at all close.

**Comment by brittonsmith** [comment]
Ah, sure, right! This is loading a single dataset and calculating, say, the radial velocity for a sphere, then changing the normal and calculating it again. Here's an example below:
```
import numpy as np
import yt

ds = yt.load("DD0046/DD0046")

sp = ds.sphere(ds.domain_center, 0.2)
print (sp['velocity_spherical_radius'])

sp.set_field_parameter('normal', np.ones(3) / np.sqrt(3))
sp.clear_data()
print (sp['velocity_spherical_radius'])
```
This gives:
```
[-14111561.13148242 -15307975.99893196 -12899747.19029205 ...
   2581133.91582618   4770170.78048369   6544593.17199936] cm/s
[ -8491218.10477266 -12443362.16458249  -6273167.96971074 ...
  -1141804.32237623   1094212.46564677   2928625.15379348] cm/s
```

**Comment by MatthewTurk** [comment]
I believe @ngoldbaum tracked this down to `get_ortho_basis`.  Is it possible to use `ortho_find` instead?

**Comment by ngoldbaum** [comment]
I fixed `get_ortho_basis`:

```
diff --git a/yt/utilities/math_utils.py b/yt/utilities/math_utils.py
index 19ae65c4c..0067297f7 100644
--- a/yt/utilities/math_utils.py
+++ b/yt/utilities/math_utils.py
@@ -1189,10 +1189,15 @@ def rotation_matrix_to_quaternion(rot_matrix):
     return np.array([w, x, y, z])

 def get_ortho_basis(normal):
-    xprime = np.cross([0.0,1.0,0.0],normal)
-    if np.sum(xprime) == 0: xprime = np.array([0.0, 0.0, 1.0])
-    yprime = np.cross(normal,xprime)
-    zprime = normal
+    # this vector has to have a non-zero dot product with normal
+    if normal[0] == 0 and normal[1] == 0:
+        u = [0, -1, 0]
+    else:
+        u = [-normal[1], normal[0], normal[2]]
+    u = np.array(u)
+    xprime = np.cross(normal, u)
+    yprime = np.cross(normal, xprime)
+    zprime = np.array(normal)
     return (xprime, yprime, zprime)

 def get_sph_r(coords):
```

But with Britton's test script I still get different results between the two outputs:

```
[ 8374686.18806114  7626242.76628462  6973383.11945249 ...
 -1676925.67006871  2047154.52077452  2396339.56737922] cm/s
[ 8534890.2357278   7775346.52416121  7105266.39459691 ...
 -1375774.46622     2118589.42470928  2534422.69143782] cm/s
```

Although the differences are smaller now...

**Comment by brittonsmith** [comment]
@MatthewTurk turned me onto a function in `math_utils.py` called `ortho_find`, which works really nicely. It seems to pass all the tests I can throw at it. I'm adding a few more now and will issue a PR shortly.

**Comment by chummels** [comment]
I'm glad that little function could be of use--I wrote that a few years ago and it was sort of a pain to create the algorithm for it.   Finding orthonormal vectors is a common task for me these days.   I thought I was the only one who used it, so I'm happy to hear others are too! Nice fix!

**Comment by ngoldbaum** [comment]
Fixed by #2013
